packages/ui/modals/Service.tsx

```diff
@@ -1,14 +1,198 @@
+import { Title, Text, Stack, Group, Box, createStyles } from '@mantine/core'
 import { ModalSettings } from '@mantine/modals/lib/context'
 import { ApiOutput } from '@weareinreach/api'
+import { useTranslation } from 'react-i18next'
 
+import { Badge, BadgeGroup, type CommunityTagProps } from '~ui/components/core'
+import { Icon } from '~ui/icon'
 import { trpc as api } from '~ui/lib/trpcClient'
 
 import { ModalTitle, ModalTitleProps } from './ModalTitle'
 
+const useStyles = createStyles((theme) => ({
+	modalBody: {
+		paddingTop: '40px',
+	},
+	sectionDivider: {
+		backgroundColor: theme.other.colors.primary.lightGray,
+		padding: 12,
+	},
+	ul: {
+		margin: 0,
+	},
+}))
+
+const WEEKDAYS = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
+
+const getHoursAndMinutes = (date: Date) => `${date.getHours()}:${date.getMinutes()}`
+
 export const ServiceModalBody = ({ serviceId }: ServiceModalBodyProps) => {
 	const { data, status } = api.service.byId.useQuery({ id: serviceId })
+	const { t } = useTranslation()
+	const { classes } = useStyles()
+
+	const SectionDivider = ({ title, children }: sectionDividerProps) => (
+		<Stack>
+			<Box className={classes.sectionDivider}>
+				<Title order={3} fw={600}>
+					{t(`service-${title}`)}
+				</Title>
+			</Box>
+			{children}
+		</Stack>
+	)
+
+	const Subsection = ({ title, children }: subsectionProps) => (
+		<Stack>
+			{title && <Title order={3}>{t(`service-${title}`)}</Title>}
+			<div>{children}</div>
+		</Stack>
+	)
+
+	const UL = (props: { children: string[] }) => {
+		const listItems = props.children.map((text) => (
+			<li key={text}>
+				<Text>{text}</Text>
+			</li>
+		))
+		return <ul className={classes.ul}>{listItems}</ul>
+	}
+
+	if (data && status === 'success') {
+		const { services, phones, emails, hours, accessDetails, attributes } = data
+
+		const allServices = services.map(({ tag }) => (
+			<Badge key={tag.tsKey} variant='service' tsKey={tag.tsKey} />
+		))
+
+		const directPhones = phones.map(({ phone }) => phone.number).join(' / ')
+
+		const directEmail = emails.find(({ email }) => {
+			email.primary
+		})
+
+		// const directWebsite = missing data for website
+
+		const serviceHours = hours.map(({ start, end, dayIndex }, i) => {
+			const formatStart = getHoursAndMinutes(start)
+			const formatEnd = getHoursAndMinutes(end)
+			const formatTime = `${WEEKDAYS[dayIndex]} ${formatStart}-${formatEnd}`
+			return <Text key={i}>{formatTime}</Text>
+		})
```

Comment:
For the operating hours have a look at how I did it [here](https://github.com/weareinreach/InReach/blob/272588700cdf81dd2c55dda3e5471e9b7ef1f71f/packages/ui/components/sections/VisitCard.tsx#L34-L69). The formatting needs to be able to change based on the currently selected locale. The Luxon library will handle the formatting of the times & the translation of the day names/abbreviations.

Reply:
Thank you! I ended up using basically the same code from your implementation. The only difference is in the [last line](https://github.com/weareinreach/InReach/blob/272588700cdf81dd2c55dda3e5471e9b7ef1f71f/packages/ui/components/sections/VisitCard.tsx#L68), where I push a different component.